packages/eslint-plugin/tools/generate-rules-table.ts

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env ts-node
+
+import fs from 'fs';
+import path from 'path';
+
+import rules from '../src/rules';
+
+const rulesData = Object.entries(rules);
+
+function createRuleLink(ruleName: string): string {
+  return `[\`@typescript-eslint/${ruleName}\`](./docs/rules/${ruleName}.md)`;
+}
+
+const emojiKey = {
+  recommended: ':heavy_check_mark:',
+  fixable: ':wrench:',
+  requiresTypeChecking: ':thought_balloon:',
+};
+
+const returnEmojiOrSpaces = <TKey extends keyof typeof emojiKey>(
+  key: TKey,
+  obj: { [K in TKey]?: unknown },
+): string => (obj[key] ? emojiKey[key] : ''.padEnd(emojiKey[key].length));
+
+const joinWithPipes = (columns: string[]): string =>
+  [' ', ...columns, ' '].join(' | ').trim();
+
+const ruleColumns = rulesData
+  .filter(([, rule]) => rule.meta.deprecated !== true)
+  .sort(([ruleNameA], [ruleNameB]) => ruleNameA.localeCompare(ruleNameB))
+  .map(([ruleName, rule]) => [
+    createRuleLink(ruleName),
+    rule.meta.docs.description,
+    returnEmojiOrSpaces('recommended', rule.meta.docs),
+    returnEmojiOrSpaces('fixable', rule.meta),
+    returnEmojiOrSpaces('requiresTypeChecking', rule.meta.docs),
+  ]);
+
+const [nameColumnLength, descriptionColumnLength] = ruleColumns.reduce(
+  ([nameColumnSize, descColumnSize], [nameStr, descStr]) => [
+    nameStr.length > nameColumnSize ? nameStr.length : nameColumnSize,
+    descStr.length > descColumnSize ? descStr.length : descColumnSize,
+  ],
+  [0, 0],
+);
```

Comment:
maybe its better to enable prettier here instead of doing manual formatting and ofc removing prettier ignore statement from readme

Reply:
in past we was using `eslint-docs` (but at some point it was removed)

don't know why i was o leave from this project

Reply:
~Personally I'm a fan of having the tables all nicely aligned~

I completely forgot that prettier does handle aligning markdown tables.

Oh well - it was a fun little scripting exercise that made a nice break from the complexity of projects I've got on my plate :joy:

Reply:
There is one more tiny think that @bradzacher recently added (there is 2nd table) with deprecated rules